R/util_scale_computation.R

```diff
@@ -0,0 +1,90 @@
+sc.append_scales <- function(
+    survey_df,
+    scale_variables_table,
+    add_z_score = TRUE
+){
+    # survey_df — cleaned, subject-level survey dataset (e.g., from Qualtrics
+    # or Qit)  scale_variables_table contains one row per scale item and
+    # specifies information about the item including the item’s intended
+    # membership in a psychometric scale (e.g., item “gms1” is part of the
+    # growth mindset scale).
+    # add_z_score — boolean value indicating whether z-score version of each
+    # scale should be computed
+
+    # take all the unique values of the scale column of scale_variables_table,
+    # and for each value, look up the column name associated with it in
+    # survey_variables_table.
+
+    # filter any non-scale variables from scale_variables_table
+    scale_variables_table <- filter(
+        scale_variables_table,
+        !is.na(scale)
+    )
+
+    scales <- unique(scale_variables_table$scale)
+
+    for(scale in scales){
+
+        # Find all the column names in survey_df corresponding to these
+        # var_names
+        scale_items <- scale_variables_table$var_name[
+            scale_variables_table$scale %in% scale
+        ]
+
+        # look up the corresponding column names in survey_df
+        survey_scale_column_names <- names(survey_df)[
+            names(survey_df) %in% scale_items
+        ]
+
+        missing_items <- scale_items[
+                !scale_items %in% survey_scale_column_names
+            ]
+
+        # If any scale items are present in survey_df, compute row means and
+        # append to survey_df.
+        # If none are present, throw a warning and do nothing.
+        if(length(survey_scale_column_names) > 0){
+            # save the existing column names to use for renaming variables
+            # later
+            old_names <- names(survey_df)
+            # compute and append row means
+            scale_means <- util.row_means(survey_df[survey_scale_column_names])
+            survey_df <- data.frame(
+                survey_df,
+                new_scale = scale_means
+            )
```

Comment:
why not just `survey_df[[new_scale]] <- util.row_means(survey_df[survey_scale_column_names])`